main/Main.hs

```diff
@@ -82,7 +99,15 @@ connectionWorker mainTid pool schema refDbStructure refIsWorkerOn = do
             atomicWriteIORef refIsWorkerOn False
             putStrLn ("Connection successful" :: Text)
 
--- | Connect to pg server if it fails retry with capped exponential backoff until success
+{-|
+  Only used by 'connectionWorker' to check if the provided db-uri lets
```

Comment:
Why is valuable to comment the scope of the function(Only used by..), I think that's better expressed by the code.